bin/generate-cli-commands.js

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env node
+'use strict';
+
+var fs                  = require('fs');
+var path                = require('path');
+var omelette            = require('omelette');
+var generateCliCommands = require('../lib/utilities/cli-command-generation');
+
+generateCliCommands();
+
+var complete = omelette('ember');
+var initFile = complete.getDefaultShellInitFile();
+var content = fs.readFileSync(initFile).toString();
+
+if (!content.match(/\.ember\/completion\.sh/)) {
+
+  complete.setupShellInitFile = function(initFile) {
+    var completionPath, programFolder, template;
+    if (initFile == null) {
+      initFile = this.getDefaultShellInitFile();
+    }
+    template = (function(_this) {
+      return function(command) {
+        return '\n# begin ' + _this.program + ' completion\n' + command + '\n# end ' + _this.program + ' completion\n';
+      };
+    })(this);
+    switch (this.shell) {
+      case 'bash':
+        programFolder = path.join(this.HOME, '.' + this.program);
+        completionPath = path.join(programFolder, 'completion.sh');
+        if (!fs.existsSync(programFolder)) {
+          fs.mkdirSync(programFolder);
+        }
+        fs.writeFileSync(completionPath, this.generateCompletionCode());
+        fs.appendFileSync(initFile, template('source ' + completionPath));
+        break;
+      case 'zsh':
+        fs.appendFileSync(initFile, template('. <(' + this.program + ' --completion)'));
+    }
```

Comment:
as a `fish` user I would love to have `autocomplete` support for it as well. If you don't have time, I can investigate this myself at a later date.